src/jsonschema.rs

```diff
@@ -68,17 +75,91 @@ struct Tag {
     extra: HashMap<String, Value>,
 }
 
-struct JSONSchema {
-    data: Tag,
-}
+impl Tag {
+    fn get_type(&self) -> Type {
```

Comment:
This method is limiting because it doesn't cover cases such as `oneOf` and `allOf`.